public/app/features/explore/Explore.tsx

```diff
@@ -315,8 +319,8 @@ function mapStateToProps(state: StoreState, { exploreId }: ExploreProps) {
 
   const { datasource, queries, range: urlRange, mode: urlMode, ui } = (urlState || {}) as ExploreUrlState;
   const initialDatasource = datasource || store.get(lastUsedDatasourceKeyForOrgId(state.user.orgId));
-  const initialQueries: DataQuery[] = ensureQueries(queries);
-  const initialRange = urlRange ? getTimeRangeFromUrl(urlRange, timeZone).raw : DEFAULT_RANGE;
+  const initialQueries: DataQuery[] = ensureQueriesMemoized(queries);
```

Comment:
Tried to optimise this a bit. The issue I think is not that much a react rendering itself, rendering the 1k is few mills, but it seems like in between those 2s that it is buffering, there is still some stuff that rerender about 100 times (still only few mills per render but it probably adds up). This stops some of the not required rerendering but still some more components that probably have similar issue.